src/plugins/interactive_setup/server/routes/configure.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { first } from 'rxjs/operators';
+
+import { schema } from '@kbn/config-schema';
+
+import type { RouteDefinitionParams } from '.';
+import { ElasticsearchConnectionStatus } from '../../common';
+import type { AuthenticateResult } from '../elasticsearch_service';
+
+export function defineConfigureRoute({
+  router,
+  logger,
+  kibanaConfigWriter,
+  elasticsearch,
+  preboot,
+}: RouteDefinitionParams) {
+  router.post(
+    {
+      path: '/internal/interactive_setup/configure',
+      validate: {
+        body: schema.oneOf([
+          schema.object({
+            hosts: schema.arrayOf(schema.uri({ scheme: 'https' }), {
+              minSize: 1,
+            }),
+            username: schema.maybe(
+              schema.string({
+                validate: (value) => {
+                  if (value === 'elastic') {
+                    return (
+                      'value of "elastic" is forbidden. This is a superuser account that can obfuscate ' +
+                      'privilege-related issues. You should use the "kibana_system" user instead.'
+                    );
+                  }
+                },
+              })
+            ),
+            password: schema.maybe(schema.string()),
```

Comment:
__nit:__ we should also require `password` if `username` is specified and vice versa. I _believe_ we can do this with conditional schema, but if not we can validate in the top-level custom `validate` function.